activerecord/lib/active_record/migration.rb

```diff
@@ -1246,7 +1269,7 @@ def migrated
     end
 
     def load_migrated
-      @migrated_versions = Set.new(Base.connection.migration_context.get_all_versions)
+      @migrated_versions = Set.new(@schema_migration.all_versions.map(&:to_i))
```

Comment:
Maybe we should move this `to_i` to inside the `SchemaMigration` class?

Reply:
Moved `to_i` to SchemaMigration class